src/Formats/ProtobufSerializer.cpp

```diff
@@ -3453,13 +3453,33 @@ namespace
                     const auto & tuple_data_type = assert_cast<const DataTypeTuple &>(*data_type);
                     size_t size_of_tuple = tuple_data_type.getElements().size();
 
-                    if (tuple_data_type.haveExplicitNames() && field_descriptor.message_type())
+                    if (field_descriptor.message_type())
```

Comment:
```suggestion
                    if (auto * message_type = field_descriptor.message_type())
```
...and further all `field_descriptor.message_type()` can be replaced with `message_type`